pysimplesoap/client.py

```diff
@@ -142,16 +140,37 @@ def __init__(self, location=None, action=None, namespace=None,
             f.close()
         self.cacert = cacert
 
-        # Create HTTP wrapper
-        Http = get_Http()
-        self.http = Http(timeout=timeout, cacert=cacert, proxy=proxy, sessions=sessions)
+        # if username and password add basic authentication
         if username and password:
-            if hasattr(self.http, 'add_credentials'):
-                self.http.add_credentials(username, password)
-        if cert and key_file:
-            if hasattr(self.http, 'add_certificate'):
-                self.http.add_certificate(key=key_file, cert=cert, domain='')
+            auth_string = "{0}:{1}".format(username, password)
+            auth_header = make_headers(basic_auth=auth_string)
+            self.http_headers.update(auth_header)
+
+        # set built-in http wrapper if specified
+        if transport:
+            set_http_wrapper(library=transport)
+
+        # else set custom http wrapper if specified
+        elif custom_transport:
+            set_http_wrapper(custom=custom_transport)
 
+        # Create HTTP wrapper
+        Http = get_Http()
+        if transport == 'urllib3':
+            self.http = Http(
+                num_pools=num_pools,
+                maxsize=maxsize,
+                timeout=timeout,
+                retries=retries,
+                headers=self.http_headers)
+        elif custom_transport:
+            self.http = Http
+        else:
+            self.http = Http(timeout=timeout, cacert=cacert, proxy=proxy, sessions=sessions)
+            if cert and key_file:
+                if hasattr(self.http, 'add_certificate'):
+                    self.http.add_certificate(key=key_file, cert=cert, 
```

Comment:
? urllib3 transport should support certificates. What's being guarded against here?